src/ui/providers/ApiKeyTreeProvider.ts

```
@@ -0,0 +1,481 @@
+import { ApiKeyTreeItem, CONTEXT_VALUES, ICONS } from '../types/TreeViewTypes';
+import { uiEventManager } from '../core/UIEventManager';
+import { uiStateManager } from '../core/UIStateManager';
+import { DataTransformers } from '../utils/DataTransformers';
+import { accessibilityService } from '../core/AccessibilityService';
+import ApiKeyManager from '../../server/core/ApiKeyManager';
+
+/**
+ * Tree data provider for API Keys view
+ */
+export class ApiKeyTreeProvider implements vscode.TreeDataProvider<ApiKeyTreeItem> {
+  private _onDidChangeTreeData: vscode.EventEmitter<ApiKeyTreeItem | undefined | null | void> = new vscode.EventEmitter<ApiKeyTreeItem | undefined | null | void>();
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)